apps/web/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {
+  FC,
+  forwardRef,
+  ReactElement,
+  Ref,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react'
+import { Button, Box, Stack, Typography, Icon } from '@island.is/island-ui/core'
+import {
+  useDialogState,
+  Dialog,
+  DialogDisclosure,
+  DialogBackdrop,
+  DialogProps,
+} from 'reakit/Dialog'
+
+import * as styles from './Modal.treat'
+
+interface ModalProps {
+  title?: string
+  label?: string
+  baseId: string
+  animated?: boolean
+  role?: 'dialog' | 'alertdialog'
+  disclosure?: ReactElement
+  onConfirm?: () => void
+  onClose?: () => void
+  buttonTextConfirm?: string
+  buttonTextCancel?: string
+}
+
+export const ModalDiv = forwardRef(
+  (props: DialogProps, ref: Ref<HTMLDivElement>) => {
+    const [mounted, setMounted] = useState(false)
+
+    useLayoutEffect(function() {
+      setMounted(true)
+    }, [])
+
+    return mounted && <div className={styles.backdrop} {...props} ref={ref} />
+  },
+)
+
+export const Modal: FC<ModalProps> = ({
```

Comment:
I am wondering why we have no modal in the island-ui kit

Reply:
I will probably move this eventually to the island-ui kit but for now I wanted to test it out for this small use case where it shows an alert dialog when switching to english.